scripts/get_max_locators.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+readonly THIS_DIR="$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd | xargs realpath)"
+readonly ROOT_DIR="${THIS_DIR}/.."
+readonly INTERNAL_DIR="${ROOT_DIR}/internal"
+
+function process_directory() {
+    local path="$1"
+
+    local dir=$(basename "${path}")
+    local parent_dir=$(basename "$(dirname "${path}")")
+
+    local largestLocator=$(grep -RoP '("\-[0-9]{3,4}")' "${path}" 2>/dev/null | sed 's/^.*"\(-[0-9]\{3,4\}\)".*$/\1/' | sort | uniq | tail -n 1)
```

Comment:
See `documentation/development.md` on advice on grep flags (we want to use `-E` over `-P`).
You may also want to use `-h` in combination with `-o` (which you already use).